### ISSUE TITLE
Add support for Telegram contacts

### ISSUE THREAD

**Issue by cyberpunkrocker-zero** [SimpleMobileTools/Simple-Contacts#640]
Please add support for Telegram contacts, like the official Google Contacts has.
Now the Simple Contacts seems to have only integration with Signal and Whatsapp contacts. Most of my friends are using Telegram, though, so those contacts imported from Google Contacts are shown separately after the 'normal' contacts.

**Comment by tibbi** [comment]
what app version are you using?

**Comment by cyberpunkrocker-zero** [comment]
Simple Contacts Pro 6.13.1

**Comment by tibbi** [comment]
it works just fine to me, what exactly is your issue? You dont see an icon at Telegram contact source at the bottom?

**Comment by cyberpunkrocker-zero** [comment]
Well, yes and no. There are two separate groups of contacts: First, the ones with phone numbers, and Signal and Whatsapp contacts (if any), and then there are the same people in a second group with only Telegram contact info (or those who have a Telegram account). The only contact entry where both groups are combined is my own contact info entry.
I suppose the Simple Contacts had some trouble importing the contact info when I installed it. How can I recombine those to groups into one? That would solve the problem.
I hope I managed to describe the issue understandably :)

**Comment by tibbi** [comment]
I dont know what you mean, Telegram contacts work just fine to me. But contact handling on Android under the hood is a ridiculous mess, so I admit that there could still be issues in some cases. Can you maybe show on a screenshot what you mean?

**Comment by cyberpunkrocker-zero** [comment]
Well I finally figured what was wrong... On the Telegram contacts the names were for some reason stored so that the whole name was in the 'first name'-field, and the 'last name'-field was empty. I had to manually edit every one of those contacts so the names were stored correctly. After that Simple Contacts showed all the contact info correctly.
I wonder how the Telegram contacts originally ended up like that?
